File: A3/Sem5/LFTC/ciordales/Labs/Laborator 3/src/main/java/FIP.java
import java.util.ArrayList;
import java.util.List;

public class FIP {

    class ItemFIP {
        private Integer atomCode;
        private Integer indexTS;

        public ItemFIP(Integer atomCode, Integer indexTS) {
            this.atomCode = atomCode;
            this.indexTS = indexTS;
        }

        @Override
        public String toString() {
            return "ItemFIP{" +
                    "atomCode='" + atomCode + '\'' +
                    ", indexTS=" + indexTS +
                    '}';
        }
    }

    private List<ItemFIP> listFIP;

    public FIP(){
        listFIP = new ArrayList<ItemFIP>();
    }

    public void addEntry(Integer atomCode, Integer indexTS){
        listFIP.add(new ItemFIP(atomCode, indexTS));
    }

    @Override
    public String toString() {
        String text = "";
        for(ItemFIP itemFIP: listFIP){
            String line = "" + itemFIP.atomCode + " " + itemFIP.indexTS + "\n";
            text = text + line;
        }
        return text;
    }
}

File: A3/Sem5/LFTC/ciordales/Labs/Laborator6/Laborator6/fip.cpp
#include "fip.h"

void FIP::addEntry(int atomCode, int indexTS) {
	listFIP.push_back({ atomCode, indexTS });
}

string FIP::toString() {
    string text = "";
    for (const pair<int,int>& itemFIP : listFIP) {
        string line = "" + to_string(itemFIP.first) + " " + to_string(itemFIP.second) + "\n";
        text = text + line;
    }
    return text;
}